src/osqp_api.c

```diff
@@ -347,35 +353,37 @@ c_int osqp_solve(OSQPWorkspace *work) {
 #endif /* ifdef CTRLC */
 
   // Initialize variables (cold start or warm start depending on settings)
-  if (!work->settings->warm_start) cold_start(work);  // If not warm start ->
-                                                      // set x, z, y to zero
+  if (!solver->settings->warm_start) osqp_cold_start(solver);  // If not warm start ->
+                                                               // set x, z, y to zero
 
   // Main ADMM algorithm
-  for (iter = 1; iter <= work->settings->max_iter; iter++) {
+
+  for (iter = 1; iter <= solver->settings->max_iter; iter++) {
```

Comment:
Maybe we should include `c_int max_iter = solver->settings->max_iter;` to avoid dereference operators in the loop condition.

Reply:
Modified as suggested.

Reply:
fixed.